src/protocol/resp/src/message/array.rs

```diff
@@ -10,6 +10,12 @@ pub struct Array {
     pub(crate) inner: Option<Vec<Message>>,
 }
 
+impl Array {
+    pub fn nil() -> Self {
```

Comment:
nit: add `*-1\r\n` in a comment as the serialized version, similarly I feel adding `Null array` as a comment to line 31/32 improves readability.